master/internal/api_experiment.go

```diff
@@ -318,24 +321,19 @@ func (a *apiServer) DeleteExperiment(
 		return nil, err
 	}
 
-	switch exists, eErr := a.m.db.ExperimentHasCheckpointsInRegistry(int(req.ExperimentId)); {
-	case eErr != nil:
-		return nil, errors.New("failed to check model registry for references")
-	case exists:
-		return nil, status.Errorf(
-			codes.InvalidArgument, "checkpoints are registered as model versions")
-	}
+	results, _, err := exputil.DeleteExperiments(ctx, a.m.system,
+		[]int32{req.ExperimentId}, nil)
 
-	if !model.ExperimentTransitions[e.State][model.DeletingState] {
-		return nil, fmt.Errorf("cannot delete experiment in %s state", e.State)
+	if err == nil {
+		if len(results) == 0 {
+			return nil, errors.Errorf("unknown error during delete query.")
+		} else if results[0].Error != nil {
+			return nil, results[0].Error
+		}
 	}
```

Comment:
did we throw away from useful errors here by accident?

Reply:
you're right, I should structure it

```
if err != nil { /* report error from the experiment selection code */ } 
else { /* current code which reports error on the individual results */ }
```